coffea/nanoevents/methods/systematics/UpDownSystematic.py

```diff
@@ -0,0 +1,71 @@
+import awkward
+from copy import copy
+from coffea.nanoevents.methods.base import behavior, Systematic
+
+
+@awkward.mixin_class(behavior)
+class UpDownSystematic(Systematic):
+    """An example instance of a simple systematic with only up/down variations."""
+
+    _udmap = {"up": 0, "down": 1}
+
+    def _build_variations(self, name, what, varying_function, *args, **kwargs):
+        whatarray = (
+            self[what] if what != "weight" else self["__systematics__", "__ones__"]
+        )
+
+        self["__systematics__", f"__{name}__"] = awkward.virtual(
+            varying_function,
```

Comment:
I'm worried it will be very hard for user-provided functions to not reference the array this virtual array is being attached to, and hence memory leaks.